src/installer.redscript.ts

```diff
@@ -145,68 +179,51 @@ export const testForRedscriptMod: VortexWrappedTestSupportedFunc = (
 // Install the Redscript stuff, as well as any archives we find
 export const installRedscriptMod: VortexWrappedInstallFunc = async (
   api: VortexApi,
-  log: VortexLogFunc,
-  files: string[],
+  _log: VortexLogFunc,
+  _files: string[],
   fileTree: FileTree,
   destinationPath: string,
 ): Promise<VortexInstallResult> => {
   // We could get a lot fancier here, but for now we don't accept
   // subdirectories anywhere other than in a canonical location.
 
-  // .\*.reds
-  // eslint-disable-next-line no-underscore-dangle
-  const hasToplevelReds = dirWithSomeIn(FILETREE_ROOT, matchRedscript, fileTree);
-  const toplevelReds = hasToplevelReds
-    ? filesUnder(FILETREE_ROOT, Glob.Any, fileTree)
-    : [];
+  const modName = makeSyntheticName(destinationPath);
 
-  // .\r6\scripts\*.reds
-  // eslint-disable-next-line no-underscore-dangle
-  const hasBasedirReds = dirWithSomeIn(
-    REDS_MOD_CANONICAL_PATH_PREFIX,
-    matchRedscript,
+  const allInstructionSets: LayoutToInstructions[] = [
+    redscriptToplevelLayout,
+    redscriptBasedirLayout,
+    redscriptCanonLayout,
+  ];
+
+  const allInstructionsPerLayout = useAllMatchingLayouts(
```

Comment:
`useAllMatchingLayouts` works right for the canon + base case because they are exclusive themselves, but toplevel will match with both of them, I think? Not that that's likely to be an issue in practice, but you could use `useFirstMatchingLayoutForInstructions` to ensure only the first match is picked (or then make toplevel layout exclusive too).

(And I guess those two should either both be with `ForInstructions` or not, but :P)

Reply:
going with a simple solution here. generally I don't think the case of a topLevel with the other two will ever come up, but eh... might as well be safe than sorry.